src/sage/rings/polynomial/multi_polynomial.pyx

```diff
@@ -416,7 +416,7 @@ cdef class MPolynomial(CommutativePolynomial):
         del Z[ind]
 
         # Make polynomial ring over all variables except var.
-        S = R.base_ring()[tuple(Z)]
+        S = PolynomialRing(R.base_ring(), Z)
```

Comment:
I think that using `__getitem__` to define a polynomial ring in library code is generally a very bad idea.  I found a few other occurrences with a very naive grep, possibly one could find more by inserting a print statement in `__getitem__` and running the test suite (and work very hard).  @fchapoton, do you agree?
```
grep -r --include=*.{py,pyx} --color -nH --null -e "ring\[" *
grep -r --include=*.{py,pyx} --color -nH --null -e "R\['" *
```
Here goes:
```
algebras/hecke_algebras/cubic_hecke_algebra.py:962:        pol_bas_ring = base_ring['h']
combinat/finite_state_machine_generators.py:1276:            polynomial_left = base_ring[var](left_side.operands()[0])
combinat/finite_state_machine_generators.py:1331:            polynomial_right = base_ring[var](next_function.operands()[0])
dynamics/arithmetic_dynamics/projective_ds.py:5769:                T = base_ring['k']
interfaces/sympy.py:195:    return base_ring[variables]
interfaces/sympy.py:227:    R = base_ring[variables]
matrix/matrix_space.py:1111:            return self.__change_ring[R]
matrix/matrix_space.py:1117:        self.__change_ring[R] = M
matrix/matrix2.pyx:3675:        R = self._base_ring[var]    # polynomial ring over the base ring
modular/quasimodform/ring.py:268:        self.__polynomial_subring = self.__modular_forms_subring[name]
schemes/elliptic_curves/hom_velusqrt.py:888:        R, Z = self._internal_base_ring['Z'].objgen()
schemes/elliptic_curves/hom_velusqrt.py:1134:        S = self._internal_base_ring['x,y']
schemes/elliptic_curves/hom_velusqrt.py:1171:        S = self._internal_base_ring['x']
schemes/elliptic_curves/ell_field.py:1604:            return R['x'].one()
schemes/elliptic_curves/ell_generic.py:3399:                R = RR['x']
schemes/elliptic_curves/hom_velusqrt.py:902:                V = R['V'].gen()
dynamics/arithmetic_dynamics/projective_ds.py4575:                    T = R['t']
dynamics/arithmetic_dynamics/projective_ds.py4932:                            T = R['t']
functions/transcendental.py633:                f = PolynomialRealDense(R['x'], coeffs)
```

Reply:
I definitely agree. It is syntactic sugar. Let’s change all of these in the library.